libcaf_core/caf/policy/work_stealing.hpp

```diff
@@ -85,7 +85,8 @@ class work_stealing : public unprofiled {
       return nullptr;
     }
     // roll the dice to pick a victim other than ourselves
-    size_t victim = d(self).rengine() % (p->num_workers() - 1);
+    std::uniform_int_distribution<size_t> uniform(0, p->num_workers() - 2);
```

Comment:
> @mavam It's a trick to avoid collision. See #392 

I see that this could be an issue. Instead of repeatedly resampling from the same distribution (which #392 changed), another approach would be to fix the candidate check. Say if `i` is the index of self, then upon collision one simple way would be to choose `i + 1 % N`.

To be frank, I would really like to measure this to see the concrete effects, but I lack the cycles to do so.